core/services/relay/evm/evm.go

```diff
@@ -691,63 +691,55 @@ func (r *Relayer) NewCCIPExecProvider(ctx context.Context, rargs commontypes.Rel
+	}
+	if !relayConfig.EffectiveTransmitterID.Valid {
+		return nil, pkgerrors.New("EffectiveTransmitterID must be specified")
 	}
 	if relayConfig.LLODONID == 0 {
 		return nil, errors.New("donID must be specified in relayConfig for LLO jobs")
 	}
+	// ensure that child loggers are namespaced by job ID which ought to be globally unique
+	lggr := r.lggr.Named(fmt.Sprintf("job-%d", rargs.JobID)).With("donID", relayConfig.LLODONID, "transmitterID", relayConfig.EffectiveTransmitterID.String)
+
 	switch relayConfig.LLOConfigMode {
 	case types.LLOConfigModeMercury, types.LLOConfigModeBlueGreen:
 	default:
 		return nil, fmt.Errorf("LLOConfigMode must be specified in relayConfig for LLO jobs (only %q or %q is currently supported)", types.LLOConfigModeMercury, types.LLOConfigModeBlueGreen)
 	}
 
 	var lloCfg lloconfig.PluginConfig
-	if err := json.Unmarshal(pargs.PluginConfig, &lloCfg); err != nil {
+	if err = json.Unmarshal(pargs.PluginConfig, &lloCfg); err != nil {
 		return nil, pkgerrors.WithStack(err)
 	}
-	if err := lloCfg.Validate(); err != nil {
+	if err = lloCfg.Validate(); err != nil {
 		return nil, err
 	}
-	relayConfig, err := relayOpts.RelayConfig()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get relay config: %w", err)
-	}
-	if relayConfig.LLODONID == 0 {
-		return nil, errors.New("donID must be specified in relayConfig for LLO jobs")
-	}
-	if relayConfig.LLOConfigMode == "" {
-		return nil, fmt.Errorf("LLOConfigMode must be specified in relayConfig for LLO jobs (can be either: %q or %q)", types.LLOConfigModeMercury, types.LLOConfigModeBlueGreen)
-	}
-	if relayConfig.ChainID.String() != r.chain.ID().String() {
-		return nil, fmt.Errorf("internal error: chain id in spec does not match this relayer's chain: have %s expected %s", relayConfig.ChainID.String(), r.chain.ID().String())
-	}
-	if !relayConfig.EffectiveTransmitterID.Valid {
-		return nil, pkgerrors.New("EffectiveTransmitterID must be specified")
-	}
 	privKey, err := r.ks.CSA().Get(relayConfig.EffectiveTransmitterID.String)
 	if err != nil {
 		return nil, pkgerrors.Wrap(err, "failed to get CSA key for mercury connection")
 	}
 
 	var transmitter LLOTransmitter
 	if lloCfg.BenchmarkMode {
-		r.lggr.Info("Benchmark mode enabled, using dummy transmitter. NOTE: THIS WILL NOT TRANSMIT ANYTHING")
-		transmitter = bm.NewTransmitter(r.lggr, fmt.Sprintf("%x", privKey.PublicKey))
+		lggr.Info("Benchmark mode enabled, using dummy transmitter. NOTE: THIS WILL NOT TRANSMIT ANYTHING")
+		transmitter = bm.NewTransmitter(lggr, fmt.Sprintf("%x", privKey.PublicKey))
 	} else {
 		clients := make(map[string]grpc.Client)
 		for _, server := range lloCfg.GetServers() {
 			var client grpc.Client
 			switch r.mercuryCfg.Transmitter().Protocol() {
 			case config.MercuryTransmitterProtocolGRPC:
 				client = grpc.NewClient(grpc.ClientOpts{
-					Logger:        r.lggr,
+					Logger:        lggr.Named(server.URL),
```

Comment:
What sort of url? Will it read OK given the names are dot separated?

Reply:
Looks like e.g. `example.com/ws` or `192.0.2.1:2345/foo/bar`

The only differentiating factor between servers is the URL, and the logger name needs to be globally unique, so this was the only way I could think of to do it.

Reply:
So the logger name needs to be unique in order to report _health_ independently, but for the purposes of just logging there is no technical conflict with sharing the log - what if we added it as a key/val? Would that be clear enough?
```suggestion
					Logger:        lggr.With("url", server.URL),
```

Reply:
Yes thats fine

Reply:
```suggestion
					Logger:        lggr.Named(fmt.Sprintf("%q", server.URL)),
```

Reply:
will do this in a followup